lib/ohai/plugins/linux/virtualization.rb

```diff
@@ -188,6 +216,18 @@ def nova_exists?
         # But, this could overwrite virtualization[:systems][:lxc] = "guest"
         # If so, we may need to look further for a differentiator (OHAI-573)
         virtualization[:systems][:lxc] = "host"
+      elsif init_environ_content =~ %r{\x00?container=lxc\x00?}
```

Comment:
Ordering wouldn't be guaranteed. And the `\x00?` is really not useful. Any character can appear before / after `container=lxc` and the regex will match.

So, either remove `\x00?` or match the boundaries properly.